### PR TITLE
[SPARK-38756][CORE][DOCS] Clean up unused security configuration items in `TransportConf`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
There are some configuration items in `TransportConf` already unused after [Update Spark key negotiation protocol](https://github.com/apache/spark/commit/3b0dd14f1c5dd033ad0a6295baa288eda9dfe10a)

- `spark.network.crypto.keyFactoryAlgorithm`
- `spark.network.crypto.keyLength`
- `spark.network.crypto.ivLength`
- `spark.network.crypto.keyAlgorithm`

so this pr clean up these configuration items from `TransportConf`,  ` security.md` and relevant UT `AuthEngineSuite`

### Why are the changes needed?
Clean up unused security configuration items.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @srowen

File: common/network-common/src/test/java/org/apache/spark/network/crypto/AuthEngineSuite.java [hunk 1]
```diff
@@ -188,19 +186,6 @@ public void testMismatchedSecret() throws Exception {
     }
   }
 
-  @Test(expected = AssertionError.class)
```
Comment (LuciferYang) [inline common/network-common/src/test/java/org/apache/spark/network/crypto/AuthEngineSuite.java:191]: This UT before `Update Spark key negotiation protocol` as follows:

```scala
  @Test(expected = InvalidKeyException.class)
  public void testBadKeySize() throws Exception {
    Map<String, String> mconf = ImmutableMap.of("spark.network.crypto.keyLength", "42");
    TransportConf conf = new TransportConf("rpc", new MapConfigProvider(mconf));

    try (AuthEngine engine = new AuthEngine("appId", "secret", conf)) {
      engine.challenge();
      fail("Should have failed to create challenge message.");

      // Call close explicitly to make sure it's idempotent.
      engine.close();
    }
  }
```
however, `spark.network.crypto.keyLength` is no longer used now, so this UT is unnecessary

**Comment by srowen** [comment]
Help me understand this a bit - I checked the diff of that change, and I don't see any of those config keys' usages removed. Maybe I missed it, or they were already unused?

**Comment by LuciferYang** [comment]
> Help me understand this a bit - I checked the diff of that change, and I don't see any of those config keys' usages removed. Maybe I missed it, or they were already unused?

Yes，they were already unused

**Comment by LuciferYang** [comment]
> Help me understand this a bit - I checked the diff of that change, and I don't see any of those config keys' usages removed. Maybe I missed it, or they were already unused?

Yes，they were already unused

**Review by srowen — APPROVED** [review]
OK I agree, it was unused

**Comment by srowen** [comment]
Merged to master

**Comment by LuciferYang** [comment]
thanks @srowen

**Pull request closed without merging**